src/scorepy/events.cpp

```diff
@@ -23,73 +23,117 @@ struct region_handle
-#endif
-};
+// Userd for regions, that only have a function name, a module,  a file and a line number
+void region_begin(const std::string& function_name, const std::string& module,
+                  const std::string& file_name, const std::uint64_t line_number)
+{
+    const auto& region_name = make_region_name(module, function_name);
+    auto& region_handle = user_regions[region_name];
+
+    if (region_handle == uninitialised_region_handle)
+    {
+        SCOREP_User_RegionInit(&region_handle.value, NULL, NULL, region_name.c_str(),
+                               SCOREP_USER_REGION_TYPE_FUNCTION, file_name.c_str(), line_number);
 
-void region_end(const std::string& region_name)
+        SCOREP_User_RegionSetGroup(region_handle.value,
+                                   std::string(module, 0, module.find('.')).c_str());
+    }
+    SCOREP_User_RegionEnter(region_handle.value);
+}
+
+// Userd for regions, that have an idetifier, aka a code object id
+void region_end(const std::string& function_name, const std::string& module,
+                const std::uintptr_t& identifier)
 {
-    const auto itRegion = regions.find(region_name);
-    if (itRegion != regions.end())
+    const auto it_region = regions.find(identifier);
+    if (it_region != regions.end())
     {
-        SCOREP_User_RegionEnd(itRegion->second.value);
+        SCOREP_User_RegionEnd(it_region->second.value);
     }
     else
     {
-        static region_handle error_region;
-        static SCOREP_User_ParameterHandle scorep_param = SCOREP_USER_INVALID_PARAMETER;
-        static bool error_printed = false;
-
-        if (std::find(EXIT_REGION_WHITELIST.begin(), EXIT_REGION_WHITELIST.end(), region_name) !=
-            EXIT_REGION_WHITELIST.end())
-        {
-            return;
-        }
-
-        if (error_region.value == SCOREP_USER_INVALID_REGION)
-        {
-            SCOREP_User_RegionInit(&error_region.value, NULL, &SCOREP_User_LastFileHandle,
-                                   "error_region", SCOREP_USER_REGION_TYPE_FUNCTION, "scorep.cpp",
-                                   0);
-            SCOREP_User_RegionSetGroup(error_region.value, "error");
-        }
-        SCOREP_User_RegionEnter(error_region.value);
-        SCOREP_User_ParameterString(&scorep_param, "leave-region", region_name.c_str());
-        SCOREP_User_RegionEnd(error_region.value);
-
-        if (!error_printed)
-        {
-            std::cerr << "SCOREP_BINDING_PYTHON ERROR: There was a region exit without an enter!\n"
-                      << "SCOREP_BINDING_PYTHON ERROR: For details look for \"error_region\" in "
-                         "the trace or profile."
-                      << std::endl;
-            error_printed = true;
-        }
+        auto& region_name = make_region_name(module, function_name);
+        region_end_error_handling(region_name);
+    }
+}
+
+// Userd for regions, that only have a function name, a module
```

Comment:
```suggestion
// Used for regions, that only have a function name, a module (user regions)
```

see above for alternative